### PR TITLE
Extract data classes

### PR DESCRIPTION
**Pull request by nodh**
Move data classes for DIF and OpenID to separate modules, i.e. decoupling them from the vck logic.

### CODE REVIEW

**Comment by nodh** [comment]
@JesusMcCloud  Please have a thorough look at the gradle files, because I'm sure there are some shorthand functions that I was not aware of for all those dependency declarations.

File: openid-data-classes/src/commonMain/kotlin/at/asitplus/openid/Json.kt
Comment (n0900) [inline openid-data-classes/src/commonMain/kotlin/at/asitplus/openid/Json.kt:1]: Remove since it is replaced by `vckJsonSerializer`?

**Review by n0900 — CHANGES_REQUESTED** [review]

File: openid-data-classes/src/commonMain/kotlin/at/asitplus/openid/Json.kt
Comment (nodh) [reply openid-data-classes/src/commonMain/kotlin/at/asitplus/openid/Json.kt:1]: Can't remove, because that serializer here is used by the data classes

**Review by n0900 — APPROVED** [review]

**Comment by JesusMcCloud** [comment]
I need to double-check, but I'm pretty sure most of the dependencies can go away, as they are transitive dependencies of signum. Tomorrow afternoon is a good time for this

**Comment by JesusMcCloud** [comment]
let's keep it that way. I just created #117 to remind me of this, once everything is merged.

**Pull request merged**